docs/content/getting-started/production-best-practices.mdx

```diff
@@ -21,17 +21,53 @@ The following list outlines some guidelines and best practices for running OpenF
 
 - The server setting `OPENFGA_DATASTORE_MAX_OPEN_CONNS` should be set to be equal to your database's max connections. For example, in Postgres, you can see this value via running the SQL query `SHOW max_connections;`. If you are running multiple instances of the OpenFGA server, you should divide this setting equally among the instances. For example, if your database's `max_connections` is 100, and you have 2 OpenFGA instances, `OPENFGA_DATASTORE_MAX_OPEN_CONNS` should be set to 50 for each instance.
 
+- The `OPENFGA_DATASTORE_MAX_IDLE_CONNS` should be set to a value no greater than the maximum open connections (see the bullet point above), but it should be set sufficiently high enough to avoid having to recreate connections on each request.
+
+  If you see a lot of database connections being closed and subsequently reopened, then you should consider increasing the maximum number of idle connections.
+
+- If idle connections are getting reaped frequently, then consider increasing the`OPENFGA_DATASTORE_CONN_MAX_IDLE_TIME` to a large value. When in doubt, prioritize keeping connections around for longer rather than shorter, because doing so will drastically improve performance.
+
+## Concurrency Limits
+:::note
+Before modifying concurrency limits please make sure you've followed the guidance for [Database Recommendations](./#database-recommendations)
+:::
+
+OpenFGA queries such as Check and ListObjects can be quite database and CPU intensive in some cases. If you notice that a single request is consuming a lot of CPU or creating a high degree of database contention, then you may consider setting some concurrency limits to protect other requests from being negatively impacted by overly aggressive queries. 
+
+The following table enumerates the server's concurrency specific settings:
+
+| flag                                    | env                                           | config                           |
+|-----------------------------------------|-----------------------------------------------|----------------------------------|
+| --max-concurrent-reads-for-list-objects | OPENFGA_MAX_CONCURRENT_READS_FOR_LIST_OBJECTS | maxConcurrentReadsForListObjects |
+| --max-concurrent-reads-for-check        | OPENFGA_MAX_CONCURRENT_READS_FOR_CHECK        | maxConcurrentReadsForCheck       |
+| --resolve-node-limit                    | OPENFGA_RESOLVE_NODE_LIMIT                    | resolveNodeLimit                 |
+| --resolve-node-breadth-limit            | OPENFGA_RESOLVE_NODE_BREADTH_LIMIT            | resolveNodeBreadthLimit          |
+
+
+Determining the right values for these settings will be based on a variety of factors including, but not limited to, the database specific deployment topology, the FGA model(s) involved, and the relationship tuples in the system. However, here are some high-level guidelines:
```

Comment:
re: "the database specific deployment topology", can we be more specific here? To me this just means how many replicas of OpenFGA are running, is there something else?

Reply:
> the database specific deployment topology

This statement has nothing to do with OpenFGA servers. It refers to the deployment topology of the database.

Reply:
Ok, made it more specific